### ISSUE TITLE
[connector] [kafka] Improve assignment and seeking

### ISSUE THREAD

**Issue by brasseld** [gravitee-io/issues#6686]
When doing a GET method to retrieve Kafka records, it always fails with a 404.
We should provide a better way to manage partition and offset (and timeout) to easily retrieve records.

**Comment by gcusnieux** [comment]
We can use this postman collection for tests;
Just call a GET and a POST and see the result of the GET.

You can also test the websocket mode with the latest version of postman, just connect to ws://apim-3-13-x-gateway.cloud.gravitee.io/kafka/messages


[kafka-http-demo.json.zip](https://github.com/gravitee-io/issues/files/7674496/kafka-http-demo.json.zip)





[kafka-http-demo.json.zip](https://github.com/gravitee-io/issues/files/7674458/kafka-http-demo.json.zip)

**Comment by RubenMMSantos** [comment]
@gcusnieux is investigating a potential issue, moved to "Blocked by QA" in the meantime

**Comment by gcusnieux** [comment]
`13:35:28.853 [vert.x-eventloop-thread-1] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition
13:57:03.587 [vert.x-eventloop-thread-1] [] ERROR i.g.connector.kafka.KafkaConnector - Unexpected error while listening for a given topic for websocket
13:57:17.007 [vert.x-eventloop-thread-1] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition
13:57:19.609 [vert.x-eventloop-thread-0] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition
13:59:10.533 [vert.x-eventloop-thread-1] [] ERROR i.g.connector.kafka.KafkaConnector - Unexpected error while listening for a given topic for websocket
13:59:35.598 [vert.x-eventloop-thread-0] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition
14:00:50.797 [vert.x-eventloop-thread-0] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition
14:01:00.988 [vert.x-eventloop-thread-1] [] ERROR i.g.c.kafka.ReadDataConnection - Kafka consume unable to seek for a given partition`

**Comment by gcusnieux** [comment]
@RubenMMSantos @mouligno I created a new ticket https://github.com/gravitee-io/issues/issues/6730 for the potential issue to unblock this card because it fixes quite a few things.. The proof, you could test it !
